ambry-router/src/main/java/com/github/ambry/router/GetManager.java

```diff
@@ -220,6 +221,12 @@ void handleResponse(ResponseInfo responseInfo) {
               }
               return serverError;
             });
+    if (responseInfo.getResponse() != null) {
```

Comment:
1. Add comment to explain why this case happen?
2. Do others(PutManager, DeleteManager) need this? 
3. Can we use a setMethod to set the ByteBufInputStream? Is it possible to set this inside extractResponseAndNotifyResponseHandler?

Reply:
Looks like we still parse the content afterwards. Is my understanding correct?

Reply:
If the goal is to fully avoid deserialization, we probably need a little bit more work to avoid deserialization in GetBlobOperation: handleBody.

Let me know if you want to have a offline chat.